### PR TITLE
197 Move all V3 Controllers into a Rest Namespace

### LINKED ISSUE #197: Move all V3 Controllers into a REST namespace
Rename the namespace from `V3` to `REST`.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #197 along with some cleanup (see the commit messages).

### CODE REVIEW

**Comment by eugenk** [comment]
I think there's something wrong with Travis. Not sure what it is.

File: config/routes.rb [hunk 1]
```diff
@@ -23,166 +38,38 @@ def rake_task?(tasks = [])
...
-      controller: 'v3/repositories', action: 'show'
-  end
-
-  scope format: false, defaults: {format: :json} do
-    root to: 'v2/search#search'
 
-    # We exclude the devise routes from these rake tasks because they load the
-    # models. When loading the models, the database needs to exist, or else it
-    # throws an error.
-    unless rake_task?(%w(db:create db:migrate db:drop
```
Comment (phyrog) [inline config/routes.rb]: I think this is still needed, might also be the problem Travis has, see #69.

File: config/routes.rb [hunk 2]
```diff
@@ -43,8 +43,13 @@ def allow_double_slashes_in_routes
   allow_double_slashes_in_routes do
     scope format: false, defaults: {format: :json} do
       # Add mappings for Devise, but skip all Devise-routes
-      devise_for :users,
-        skip: %i(registrations confirmations sessions unlocks passwords)
+      # We exclude these Devise mappings from these rake tasks because they
+      # load the models. When loading the models, the database needs to exist,
+      # or else it throws an error.
+      unless rake_task?(%w(db:create db:migrate db:drop))
```
Comment (phyrog) [inline config/routes.rb]: Might be necessary to add `db:recreate` as well

Comment (eugenk) [reply config/routes.rb]: Right 😅

File: app/controllers/rest/application_controller.rb [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
+module Rest
   # This is the base class for all API Version 3 Controllers.
   class ApplicationController < ActionController::API
     extend DSL
 
     context do
       {current_user: current_user}
     end
+
+    def no_op
```
Comment (phyrog) [inline app/controllers/rest/application_controller.rb]: Wrap in `:nocov:` please

Comment (eugenk) [reply app/controllers/rest/application_controller.rb]: Better: 664c3e40

Comment (phyrog) [reply app/controllers/rest/application_controller.rb]: :+1:

**Comment by eugenk** [comment]
Well, that took some doing...

**Pull request merged**